src/material/core/theming/tests/theming-typography-hierarchy.spec.ts

```diff
@@ -0,0 +1,84 @@
+import {parse} from 'postcss';
+import {compileString} from 'sass';
+import {runfiles} from '@bazel/runfiles';
+import * as path from 'path';
+
+import {createLocalAngularPackageImporter} from '../../../../../tools/sass/local-sass-importer';
+import {pathToFileURL} from 'url';
+
+// Note: For Windows compatibility, we need to resolve the directory paths through runfiles
+// which are guaranteed to reside in the source tree.
+const testDir = path.join(runfiles.resolvePackageRelative('../_all-theme.scss'), '../tests');
+const packagesDir = path.join(runfiles.resolveWorkspaceRelative('src/cdk/_index.scss'), '../..');
+
+const localPackageSassImporter = createLocalAngularPackageImporter(packagesDir);
+
+const mdcSassImporter = {
+  findFileUrl: (url: string) => {
+    if (url.toString().startsWith('@material')) {
+      return pathToFileURL(
+        path.join(runfiles.resolveWorkspaceRelative('./node_modules'), url),
+      ) as URL;
+    }
+    return null;
+  },
+};
+
+/** Transpiles given Sass content into CSS. */
+function transpile(content: string) {
+  return compileString(
+    `
+        @use '../../../index' as mat;
+        @use '../../../../material-experimental/index' as matx;
+
+        $internals: _mat-theming-internals-do-not-access;
+
+        $theme: matx.define-theme();
+
+        ${content}
+      `,
+    {
+      loadPaths: [testDir],
+      importers: [localPackageSassImporter, mdcSassImporter],
+    },
+  ).css.toString();
+}
+
+function verifyFullSelector(css: string, selector: string) {
+  expect(css).toMatch(
+    new RegExp(String.raw`(^|\n)` + selector.replace(/\./g, String.raw`\.`) + String.raw` \{`),
+  );
+}
+
+describe('typography hierarchy', () => {
+  describe('for M3', () => {
+    it('should emit styles for h1', () => {
+      const css = transpile('@include mat.typography-hierarchy($theme)');
+      verifyFullSelector(
+        css,
+        '.mat-display-large, .mat-typography .mat-display-large, .mat-typography h1',
+      );
+    });
+
+    it('should emit default body styles', () => {
+      const css = transpile('@include mat.typography-hierarchy($theme)');
+      verifyFullSelector(css, '.mat-body-large, .mat-typography .mat-body-large, .mat-typography');
+    });
+
```

Comment:
Should we have a test where the selector is included under another class?

Reply:
Done